test/fw/ptl/lib/ptl_mom.py

```diff
@@ -383,6 +512,133 @@ def revert_to_defaults(self, delvnodedefs=True):
+        :param primary_server: object of the primary PBS server
+        :type primary_server: PBSService
+        :param vals_to_set: dict of pbs.conf values to set
+        :type vals_to_set: dict
+        """
+        '''svr_hostnames = [svr.hostname for svr in self.servers.values()]
+
+        #for mom in moms.values():
+        if self.hostname in svr_hostnames:
+            return'''
+
+        new_pbsconf = dict(vals_to_set)
+        restart_mom = False
+        pbs_conf_val = self.du.parse_pbs_config(self.hostname)
+        if not pbs_conf_val:
+            raise ValueError("Could not parse pbs.conf on host %s" %
+                             (self.hostname))
+
+        # to start with, set all keys in new_pbsconf with values from the
+        # existing pbs.conf
+        keys_to_delete = []
+        for conf in new_pbsconf:
+            if conf in pbs_conf_val:
+                new_pbsconf[conf] = pbs_conf_val[conf]
+            else:
+                # existing pbs.conf doesn't have a default variable set
+                # Try to determine the default
+                val = self._get_dflt_pbsconfval(conf,
+                                                primary_server.hostname,
+                                                "mom", self)
+                if val is None:
+                    self.logger.error("Couldn't revert %s in pbs.conf"
+                                      " to its default value" %
+                                      (conf))
+                    keys_to_delete.append(conf)
+                else:
+                    new_pbsconf[conf] = val
+
+        for key in keys_to_delete:
+            del(new_pbsconf[key])
+
+        # Set the mom start bit to 1
+        if (new_pbsconf["PBS_START_MOM"] != "1"):
+            new_pbsconf["PBS_START_MOM"] = "1"
+            restart_mom = True
+
+        # Set PBS_CORE_LIMIT, PBS_SCP and PBS_SERVER
+        if new_pbsconf["PBS_CORE_LIMIT"] != "unlimited":
+            new_pbsconf["PBS_CORE_LIMIT"] = "unlimited"
+            restart_mom = True
+        if new_pbsconf["PBS_SERVER"] != primary_server.hostname:
+            new_pbsconf["PBS_SERVER"] = primary_server.hostname
+            restart_mom = True
+        if "PBS_SCP" not in new_pbsconf:
+            scppath = self.du.which(self.hostname, "scp")
+            if scppath != "scp":
+                new_pbsconf["PBS_SCP"] = scppath
+                restart_mom = True
+        if new_pbsconf["PBS_LOG_HIGHRES_TIMESTAMP"] != "1":
+            new_pbsconf["PBS_LOG_HIGHRES_TIMESTAMP"] = "1"
+            restart_mom = True
+
+        # Check if existing pbs.conf has more/less entries than the
+        # default list
+        if len(pbs_conf_val) != len(new_pbsconf):
+            restart_mom = True
+        # Check if existing pbs.conf has correct ownership
+        dest = self.du.get_pbs_conf_file(self.hostname)
+        (cf_uid, cf_gid) = (os.stat(dest).st_uid, os.stat(dest).st_gid)
```

Comment:
I agree with what you are saying. But I have just copied the existing code present in pbs_testsuite.py file to MoM class. Since this is not currently affecting any tests in the regression, I suggest we take this improvement in a separate PR if possible.

Reply:
Please file ticket for this so we don't forget.